core/templates/pages/learner-dashboard-page/goals-tab.component.ts

```diff
@@ -43,19 +43,23 @@ export class GoalsTabComponent implements OnInit {
     private deviceInfoService: DeviceInfoService) {
   }
 
-  @Input() currentGoals: LearnerTopicSummary[];
-  @Input() editGoals: LearnerTopicSummary[];
-  @Input() completedGoals: LearnerTopicSummary[];
-  @Input() untrackedTopics: Record<string, LearnerTopicSummary[]>;
-  @Input() partiallyLearntTopicsList: LearnerTopicSummary[];
-  @Input() learntToPartiallyLearntTopics: string[];
-  @ViewChild('dropdown', {'static': false}) dropdownRef;
-  learnerDashboardActivityIds: LearnerDashboardActivityIds;
-  MAX_CURRENT_GOALS_LENGTH: number;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() currentGoals!: LearnerTopicSummary[];
+  @Input() editGoals!: LearnerTopicSummary[];
+  @Input() completedGoals!: LearnerTopicSummary[];
+  @Input() untrackedTopics!: Record<string, LearnerTopicSummary[]>;
+  @Input() partiallyLearntTopicsList!: LearnerTopicSummary[];
+  @Input() learntToPartiallyLearntTopics!: string[];
+
+  @ViewChild('dropdown', {'static': false}) dropdownRef?: ElementRef;
```

Comment:
Why `?`, we prefer not to use it, right?

Reply:
Done